test/e2e/diagram.overlays.rendering.test.ts

```diff
@@ -45,33 +44,72 @@ class ImageSnapshotThresholds extends MultiBrowserImageSnapshotThresholds {
+  return join(config.customSnapshotsDir, `on-position-${position}`);
+}
+
+async function addOverlay(bpmnContainerElementHandle: ElementHandle<Element>, bpmnElementId: string, position: OverlayPosition): Promise<void> {
+  await page.fill('#bpmn-id-input', bpmnElementId);
+  await clickOnButton(position);
+  await bpmnContainerElementHandle.waitForSelector(`svg > g > g:nth-child(3) > g[data-bpmn-id="${bpmnElementId}"]`);
+}
+
 describe('BPMN elements with overlays', () => {
   const imageSnapshotThresholds = new ImageSnapshotThresholds();
-  const imageSnapshotConfigurator = new ImageSnapshotConfigurator(imageSnapshotThresholds.getThresholds(), 'overlays', imageSnapshotThresholds.getDefault());
+  const imageSnapshotConfigurator = new ImageSnapshotConfigurator(imageSnapshotThresholds.getThresholds(), 'overlay', imageSnapshotThresholds.getDefault());
 
-  const pageTester = new PageTester({ pageFileName: 'overlays', expectedPageTitle: 'BPMN Visualization Overlays' });
-  const bpmnDiagramName = 'overlays.start.flow.task';
+  const pageTester = new PageTester({ pageFileName: 'overlay', expectedPageTitle: 'BPMN Visualization - Overlay' });
+  const bpmnDiagramName = 'overlays.start.flow.task.gateway';
 
-  it('Check badges are present on StartEvent, Task', async () => {
+  it.each([
+    <OverlayShapePosition>'top-left',
+    <OverlayShapePosition>'top-center',
+    <OverlayShapePosition>'top-right',
+    <OverlayShapePosition>'bottom-left',
+    <OverlayShapePosition>'bottom-center',
+    <OverlayShapePosition>'bottom-right',
+    <OverlayShapePosition>'middle-left',
+    <OverlayShapePosition>'middle-right',
+  ])(`add overlay on StartEvent, Gateway and Task on %s`, async (position: OverlayShapePosition) => {
     const bpmnContainerElementHandle = await pageTester.loadBPMNDiagramInRefreshedPage(bpmnDiagramName);
-    await bpmnContainerElementHandle.waitForSelector('svg > g > g:nth-child(3) > g[data-bpmn-id="StartEvent_1"]');
-    // TODO: uncomment or use when we add support for edge overlay
-    // await bpmnContainerElementHandle.waitForSelector('svg > g > g:nth-child(3) > g[data-bpmn-id="Flow_1"]');
-    await bpmnContainerElementHandle.waitForSelector('svg > g > g:nth-child(3) > g[data-bpmn-id="Activity_1"]');
+
+    await addOverlay(bpmnContainerElementHandle, 'StartEvent_1', position);
+    await addOverlay(bpmnContainerElementHandle, 'Activity_1', position);
+    await addOverlay(bpmnContainerElementHandle, 'Gateway_1', position);
 
     const image = await page.screenshot({ fullPage: true });
     const config = imageSnapshotConfigurator.getConfig(bpmnDiagramName);
-    expect(image).toMatchImageSnapshot(config);
+    expect(image).toMatchImageSnapshot({
+      ...config,
+      customSnapshotIdentifier: 'add.overlay.on.task.gateway.and.event',
+      customSnapshotsDir: buildOverlaySnapshotDir(config, position),
+    });
   });
+
+  it.skip.each([<OverlayEdgePosition>'start', <OverlayEdgePosition>'middle', <OverlayEdgePosition>'end'])(
```

Comment:
⚠️ we should load a dedicated file for flow tests with various flow paths (vertical, horizontal and complex) and length to see how the rendering behaves. To ensure we don't forget to add such use cases, we can replace `bpmnDiagramName` by `'missing.bpmn'` and the test will fail for sure when enabled.

Reply:
Yes, we should add **messageFlow** too.